src/Form/CloudFlareAdminSettingsForm.php

```diff
@@ -65,12 +144,28 @@ public function buildForm(array $form, FormStateInterface $form_state) {
    */
   public function validateForm(array &$form, FormStateInterface $form_state) {
     $email = $form_state->getValue('email');
-    $is_email_valid = \Drupal::service('email.validator')->isValid($email);
+    $apikey = $form_state->getValue('apikey');
+    $is_email_valid = $this->emailValidator->isValid($email);
+
+    // Using the bare-metal-class here for diagnostic purposes.
+    $zone_api_direct = new ZoneApi($apikey, $email);
 
     if (!$is_email_valid) {
       $form_state->setErrorByName('email', $this->t('Please enter a valid e-mail address.'));
     }
 
+    try {
```

Comment:
That's smart, so my comment about checking for creds in the check too, still stands but becomes less of a attack/crash vector now that I've seen this ;).